src/electionguard/manifest.cpp

```diff
@@ -1055,6 +1107,59 @@ namespace electionguard
     unique_ptr<ElementModQ> ContestDescription::crypto_hash() { return pimpl->crypto_hash(); }
     unique_ptr<ElementModQ> ContestDescription::crypto_hash() const { return pimpl->crypto_hash(); }
 
+    bool ContestDescription::isValid() const
```

Comment:
this validity check will verify that the data coming in meets the expectations of the rest of the library.  specifically we check things like duplicate sequence orders, that candidates actually exist, etc.